proto/minder/v1/minder.proto

```diff
@@ -1441,37 +1464,67 @@ message GetDataSourceByIdRequest {
 }
 
 message GetDataSourceByIdResponse {
-    // TODO
+    DataSource data_source = 1;
+}
+
+// GetDataSourceByNameRequest is the request message for the GetDataSourceByName RPC.
+message GetDataSourceByNameRequest {
+    ContextV2 context = 1;
+
+    string name = 2 [
+        (buf.validate.field).string = {
+            pattern: "^[A-Za-z][-[:word:]]*$",
+            max_len: 200,
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+}
+
+message GetDataSourceByNameResponse {
+    DataSource data_source = 1;
 }
 
 message ListDataSourcesRequest {
     ContextV2 context = 1;
 }
 
 message ListDataSourcesResponse {
-    // TODO
+    repeated DataSource data_sources = 1;
 }
 
 message UpdateDataSourceRequest {
     ContextV2 context = 1;
```

Comment:
Interesting, do we need the Context here since we have it as part of the DataSource?

Reply:
That's true, I missed this. Thanks